pingcap/tidb-operator
# autoscaler: refactor monitor syncing logic and crd

### What problem does this PR solve? 

The TidbMonitor may not get updated if TidbCluster has been synced but TidbMonitor failed. Also the CRD needs to separate the resources used by TiDB and TiKV

### What is changed and how does it work?

Sync TidbMonitor in a separate logic. Refactor CRD

### Check List 

Tests 

 - Unit test
 - Manual Test
Details:
Create a TidbCluster and TidbClusterAutoScaler, stress the cluster. Should see an autoscaling cluster created and can see the autoscaling cluster on Grafana
![image](https://user-images.githubusercontent.com/5250490/94405002-7ab5da00-01a2-11eb-85a9-bb3fecb86082.png)

Code changes

 - Has Go code change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
NONE
```
